src/repository.rs

```diff
@@ -317,11 +321,18 @@ where
         HttpRepository {
             url: url,
             client: client,
+            http_hook: None,
             user_agent: user_agent,
             _interchange: PhantomData,
         }
     }
 
+    /// Set an hook function to be invoked before sending an HTTP request.
+    pub fn set_http_hook(&mut self, hook: Option<HttpHook>) -> &Self {
```

Comment:
Oops. Just saw this. I'm not using getters since this is a builder and presumably people know what they're putting in before they do, so I won't need to support both.

And yeah, something similar to your test case but without the panic and with an actual manipulation.